[2021-03-03T16:40:57.872186221+02:00][INFO][13748 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(341)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.2.28/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log
[2021-03-03T16:40:57.877248947+02:00][INFO][13748 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-03T16:40:57.887822664+02:00][WARN][13748 main][defaults][Some("shared/defaults/src/lib.rs")][Some(109)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-03T16:40:57.887965635+02:00][INFO][13748 main][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(90)] config is BootstrapConfig { bootstrapper_enabled: true, api_endpoint: "https://api.tabnine.com/experiments/register/v2", registration_interval_seconds: 3600, bark_interval_seconds: 2, force_version: None }
[2021-03-03T16:40:57.888343535+02:00][INFO][13748 main][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:58.187781359+02:00][INFO][13748 thread][watchdog_server][Some("tabnine/watchdog_server/src/lib.rs")][Some(53)] watchdog is locking path: /home/denta/.config/TabNine/monitoring/watchdog.lock
[2021-03-03T16:40:58.189333410+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-03T16:40:58.232168286+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/tabnine_version_3.2.28.prefs.json")
[2021-03-03T16:40:58.232378215+02:00][WARN][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(109)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-03T16:40:58.232522786+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-03T16:40:58.232599654+02:00][WARN][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(109)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-03T16:40:58.232657033+02:00][INFO][13748 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(192)] trying to start tabnine version Some("3.2.28")
[2021-03-03T16:40:58.232734071+02:00][INFO][13748 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(267)] spawning tabnine with args ["/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.2.28/x86_64-unknown-linux-gnu/TabNine", "--client", "jupyter", "--log-file-path", "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log", "--no_bootstrap"]
[2021-03-03T16:40:58.239170060+02:00][INFO][13769 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(341)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.2.28/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log --no_bootstrap
[2021-03-03T16:40:58.239403447+02:00][INFO][13769 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(284)] inner at this point
[2021-03-03T16:40:58.240619864+02:00][INFO][13748 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(271)] started tabnine process pid 13769 version Some("3.2.28"), desired version was Some("3.2.28") this version is Some("3.2.28")
[2021-03-03T16:40:58.241161216+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-03T16:40:58.258102187+02:00][INFO][13748 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-03T16:40:58.885405320+02:00][INFO][13769 main][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:58.887546934+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(168)] lunar settings update thread started (Remote)
[2021-03-03T16:40:58.896734953+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:40:59.165725493+02:00][INFO][13769 local-forwarder][lunar_local::bridge][Some("deep/lunar_local/src/bridge.rs")][Some(75)] TabNine Local communication thread started.
[2021-03-03T16:40:59.165955707+02:00][INFO][13769 local-forwarder][lunar_local::bridge][Some("deep/lunar_local/src/bridge.rs")][Some(13)] Loading model...
[2021-03-03T16:40:59.165982157+02:00][INFO][13769 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-03T16:40:59.166117607+02:00][INFO][13769 local-forwarder][lunar_local::locker][Some("deep/lunar_local/src/locker.rs")][Some(15)] Trying to lock: "/tmp/model.lock"
[2021-03-03T16:40:59.166178921+02:00][INFO][13769 local-forwarder][lunar_local::locker][Some("deep/lunar_local/src/locker.rs")][Some(17)] Lock acquired, performing the inner action
[2021-03-03T16:40:59.166181993+02:00][INFO][13769 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-03T16:40:59.166233316+02:00][INFO][13769 local-forwarder][lunar_local::downloader][Some("deep/lunar_local/src/downloader.rs")][Some(173)] Starting download (last_failure=None).
[2021-03-03T16:40:59.166331165+02:00][WARN][13769 main][defaults][Some("shared/defaults/src/lib.rs")][Some(109)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-03T16:40:59.166393760+02:00][INFO][13769 main][frontend][Some("tabnine/frontend/src/lib.rs")][Some(82)] Started.
[2021-03-03T16:40:59.166626913+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(168)] lunar settings update thread started (Local)
[2021-03-03T16:40:59.166941844+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:40:59.167061975+02:00][INFO][13769 main][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.217447238+02:00][INFO][13769 local-forwarder][lunar_local::downloader][Some("deep/lunar_local/src/downloader.rs")][Some(723)] Tried to find cached metadata but got 'No such file or directory (os error 2)', so downloading it instead...
[2021-03-03T16:40:59.334831438+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:40:59.334995219+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:40:59.334831836+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:40:59.336328763+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:40:59.368778030+02:00][INFO][13748 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.385581475+02:00][INFO][13769 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.385937889+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:40:59.386146372+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:40:59.386847922+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:40:59.386968901+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:40:59.469305323+02:00][INFO][13748 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.486065923+02:00][INFO][13769 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.486360541+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:40:59.486562653+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:40:59.486068975+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:40:59.486949448+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:40:59.510398080+02:00][INFO][13748 thread][watchdog_server::monitor_loop][Some("tabnine/watchdog_server/src/monitor_loop.rs")][Some(227)] watching new process 13748-TabNine
[2021-03-03T16:40:59.569669373+02:00][INFO][13748 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.586483953+02:00][INFO][13769 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:40:59.586808960+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:40:59.587169292+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:40:59.587264130+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:40:59.587409081+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:40:59.802993744+02:00][INFO][13748 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(246)] current version is Some(Version { major: 3, minor: 2, patch: 28, pre: [], build: [] }), spawned version is Some(Version { major: 3, minor: 2, patch: 28, pre: [], build: [] }), remote version is Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }
[2021-03-03T16:41:00.470585218+02:00][INFO][13748 thread][watchdog_server::monitor_loop][Some("tabnine/watchdog_server/src/monitor_loop.rs")][Some(227)] watching new process 13769-TabNine
[2021-03-03T16:41:00.816855540+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-03T16:41:00.817454496+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/3.3.82_hash_16898104813460808464_bundle.prefs.json")
[2021-03-03T16:41:00.817626002+02:00][WARN][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(109)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-03T16:41:00.817796085+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(604)] lock is at "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/bundle.lock"
[2021-03-03T16:41:00.818360260+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(255)] downloading from "https://update.tabnine.com/bundles/3.3.82/x86_64-unknown-linux-musl/TabNine.zip" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine.zip"
[2021-03-03T16:41:00.818532624+02:00][INFO][13748 bs_main_rlp][bootstrapper::download][Some("tabnine/bootstrapper/src/download.rs")][Some(29)] parent path is "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu"
[2021-03-03T16:41:00.818772173+02:00][INFO][13748 bs_main_rlp][http_requests][Some("shared/http_requests/src/lib.rs")][Some(97)] Downloading from https://update.tabnine.com/bundles/3.3.82/x86_64-unknown-linux-musl/TabNine.zip
[2021-03-03T16:41:01.436239666+02:00][INFO][13769 local-forwarder][http_requests][Some("shared/http_requests/src/lib.rs")][Some(73)] Making API request to https://api.tabnine.com/model_metadata/v1
[2021-03-03T16:41:01.696597965+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(168)] lunar settings update thread started (Remote)
[2021-03-03T16:41:01.696914302+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:41:01.697278334+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(168)] lunar settings update thread started (Local)
[2021-03-03T16:41:01.697576719+02:00][INFO][13769 local-forwarder][lunar_local::bridge][Some("deep/lunar_local/src/bridge.rs")][Some(75)] TabNine Local communication thread started.
[2021-03-03T16:41:01.697836105+02:00][INFO][13769 local-forwarder][lunar_local::bridge][Some("deep/lunar_local/src/bridge.rs")][Some(13)] Loading model...
[2021-03-03T16:41:01.698034384+02:00][INFO][13769 local-forwarder][lunar_local::locker][Some("deep/lunar_local/src/locker.rs")][Some(15)] Trying to lock: "/tmp/model.lock"
[2021-03-03T16:41:01.700941975+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:41:02.104770511+02:00][INFO][13769 local-forwarder][lunar_local::downloader][Some("deep/lunar_local/src/downloader.rs")][Some(269)] Downloaded model metadata [hash=ce94127b].
[2021-03-03T16:41:02.328161091+02:00][INFO][13769 local-forwarder][lunar_local::downloader][Some("deep/lunar_local/src/downloader.rs")][Some(214)] Attempting to load deep model (attempt: #1)
[2021-03-03T16:41:02.328364660+02:00][ERROR][13769 local-forwarder][lunar_local::downloader][Some("deep/lunar_local/src/downloader.rs")][Some(228)] Loading model failed... next retry in 60 secs. failure reason: [FMA instructions not supported]
[2021-03-03T16:41:18.370945169+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:41:18.370954944+02:00][INFO][13748 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:41:18.370974002+02:00][INFO][13769 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:41:18.371161900+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:41:18.371277544+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:41:18.371023221+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:41:18.371631119+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:41:18.371678859+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:41:18.371838755+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:41:18.371948577+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:41:18.438403661+02:00][INFO][13769 update_loop][http_requests][Some("shared/http_requests/src/lib.rs")][Some(73)] Making API request to https://api.tabnine.com/version/v1
[2021-03-03T16:41:18.471512874+02:00][INFO][13748 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:41:18.471554444+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:41:18.471697136+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:41:18.471640926+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Local)
[2021-03-03T16:41:18.471669795+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(183)] lunar settings detected config modified (Remote)
[2021-03-03T16:41:18.471595814+02:00][INFO][13769 thread][updater::global_config][Some("tabnine/updater/src/global_config.rs")][Some(687)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-03T16:41:18.472280444+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:41:18.472304789+02:00][INFO][13769 lunar settings update thread (Remote)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Remote lunar settings
[2021-03-03T16:41:18.472608448+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:41:18.472618902+02:00][INFO][13769 lunar settings update thread (Local)][server::lunar_connection][Some("tabnine/server/src/lunar_connection.rs")][Some(175)] Successfully updated Local lunar settings
[2021-03-03T16:41:19.838996628+02:00][INFO][13769 update_loop][updater][Some("tabnine/updater/src/lib.rs")][Some(654)] Server did not respond with version, so using default URL.
[2021-03-03T16:41:22.316650830+02:00][INFO][13769 update_loop][updater][Some("tabnine/updater/src/lib.rs")][Some(328)] Current version is 3.2.28, remote has version 3.1.17.
[2021-03-03T16:41:24.729642134+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(289)] completed downloading to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine.zip"
[2021-03-03T16:41:24.730094714+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(404)] outpath is "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine"
[2021-03-03T16:41:38.005734883+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(404)] outpath is "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine-deep-cloud"
[2021-03-03T16:41:45.020798422+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(404)] outpath is "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine-deep-local"
[2021-03-03T16:41:52.587208236+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(404)] outpath is "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/WD-TabNine"
[2021-03-03T16:41:58.183340082+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(437)] creating root path "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu"
[2021-03-03T16:41:58.183941072+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(452)] moving from "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu"
[2021-03-03T16:41:58.184183276+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(497)] moving "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine-deep-local" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine-deep-local"
[2021-03-03T16:41:58.184389331+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(497)] moving "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/WD-TabNine" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/WD-TabNine"
[2021-03-03T16:41:58.184610782+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(497)] moving "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine"
[2021-03-03T16:41:58.184770993+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(497)] moving "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/TabNine-deep-cloud" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine-deep-cloud"
[2021-03-03T16:41:58.184935210+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(497)] moving "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/bundle.lock" to "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/bundle.lock"
[2021-03-03T16:41:58.192800424+02:00][INFO][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(298)] finished extracting
[2021-03-03T16:41:58.193748551+02:00][WARN][13748 bs_main_rlp][bootstrapper::bundle][Some("tabnine/bootstrapper/src/bundle.rs")][Some(618)] relinquishing lock at path "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/0.0.1/x86_64-unknown-linux-gnu/bundle.lock"
[2021-03-03T16:41:59.391432988+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-03T16:41:59.392273503+02:00][WARN][13748 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(304)] restart required
[2021-03-03T16:41:59.392562653+02:00][INFO][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-03T16:41:59.392667801+02:00][WARN][13748 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(109)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-03T16:41:59.392747088+02:00][INFO][13748 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(211)] got kill message from bootstrapper. Bye Bye 👋🏻...
[2021-03-03T16:41:59.742441431+02:00][INFO][13769 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(332)] here
[2021-03-03T16:41:59.742639513+02:00][INFO][13769 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(334)] exiting....
[2021-03-04T09:03:26.238298453+02:00][INFO][2164 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(38)] token path: /home/denta/.config/TabNine/tabnine.token
[2021-03-04T09:03:26.277082499+02:00][INFO][2164 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(21)] Failed to read token file, user cannot be identified: No such file or directory (os error 2)
[2021-03-04T09:03:26.277643215+02:00][WARN][2164 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(166)] No user token available
[2021-03-04T09:03:26.277795458+02:00][INFO][2164 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(434)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log
[2021-03-04T09:03:26.278904430+02:00][INFO][2164 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-04T09:03:26.310172763+02:00][WARN][2164 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T09:03:26.310306185+02:00][INFO][2164 main][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(93)] config is BootstrapConfig { bootstrapper_enabled: true, api_endpoint: "https://api.tabnine.com/experiments/register/v2", registration_interval_seconds: 3600, bark_interval_seconds: 2, force_version: None }
[2021-03-04T09:03:26.334454909+02:00][INFO][2164 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:26.597792408+02:00][INFO][2164 main][auth::user::token_watcher][Some("shared/auth/src/user/token_watcher.rs")][Some(65)] Start watching token file
[2021-03-04T09:03:26.882543881+02:00][WARN][2164 main][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(564)] Cannot run local model: FMA instructions not supported

[2021-03-04T09:03:26.883027119+02:00][INFO][2164 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(64)] Initializing watchdog client...
[2021-03-04T09:03:27.088452665+02:00][INFO][2164 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T09:03:27.090657956+02:00][INFO][2164 bs_main_autobarker][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T09:03:27.099926740+02:00][INFO][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T09:03:27.136726263+02:00][INFO][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/tabnine_version_3.3.82.prefs.json")
[2021-03-04T09:03:27.161835673+02:00][INFO][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-04T09:03:27.162050777+02:00][WARN][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T09:03:27.162203770+02:00][INFO][2164 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(199)] trying to start tabnine version Some("3.3.82")
[2021-03-04T09:03:27.162369894+02:00][INFO][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T09:03:27.162524823+02:00][INFO][2164 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(289)] spawning tabnine with args ["/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine", "--client", "jupyter", "--log-file-path", "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log", "--no_bootstrap"]
[2021-03-04T09:03:27.163325861+02:00][INFO][2164 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(293)] started tabnine process pid 2192 version Some("3.3.82"), desired version was Some("3.3.82") this version is Some("3.3.82")
[2021-03-04T09:03:27.163861294+02:00][INFO][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T09:03:27.165974493+02:00][INFO][2192 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(38)] token path: /home/denta/.config/TabNine/tabnine.token
[2021-03-04T09:03:27.166306710+02:00][INFO][2192 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(21)] Failed to read token file, user cannot be identified: No such file or directory (os error 2)
[2021-03-04T09:03:27.166413551+02:00][WARN][2192 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(166)] No user token available
[2021-03-04T09:03:27.166515655+02:00][INFO][2192 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(434)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log --no_bootstrap
[2021-03-04T09:03:27.167494365+02:00][INFO][2192 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(64)] Initializing watchdog client...
[2021-03-04T09:03:27.181570477+02:00][INFO][2164 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T09:03:27.325643571+02:00][INFO][2192 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T09:03:27.381609185+02:00][INFO][2192 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-03-04T09:03:27.382721409+02:00][INFO][2164 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-03-04T09:03:27.789931406+02:00][INFO][2164 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:27.886436455+02:00][INFO][2192 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:27.998754110+02:00][INFO][2192 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(623)] model hash updater starting
[2021-03-04T09:03:27.999490560+02:00][INFO][2192 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Local)
[2021-03-04T09:03:27.999596518+02:00][INFO][2192 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Remote)
[2021-03-04T09:03:27.999693241+02:00][INFO][2192 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Local)
[2021-03-04T09:03:27.999811815+02:00][INFO][2192 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Remote)
[2021-03-04T09:03:28.080715961+02:00][INFO][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T09:03:28.081250278+02:00][INFO][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/message_shown_timestamp_default.prefs.json")
[2021-03-04T09:03:28.081453684+02:00][WARN][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T09:03:28.081628495+02:00][INFO][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-04T09:03:28.081781760+02:00][WARN][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T09:03:28.088145210+02:00][INFO][2164 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:28.203245256+02:00][INFO][2192 thread][auth::user::token_watcher][Some("shared/auth/src/user/token_watcher.rs")][Some(65)] Start watching token file
[2021-03-04T09:03:28.205424520+02:00][INFO][2164 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:28.230536838+02:00][WARN][2192 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(564)] Cannot run local model: FMA instructions not supported

[2021-03-04T09:03:28.513934192+02:00][INFO][2192 main][frontend::initialize][Some("tabnine/frontend/src/initialize.rs")][Some(5)] Started.
[2021-03-04T09:03:28.525844147+02:00][INFO][2192 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:28.727699688+02:00][INFO][2164 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:28.727732616+02:00][INFO][2192 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T09:03:29.011340533+02:00][INFO][2164 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(260)] current version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), spawned version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), remote version is Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }
[2021-03-04T09:03:29.181508674+02:00][INFO][2164 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(332)] don't need to upgrade
[2021-03-04T09:03:29.438208384+02:00][INFO][2192 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Remote)
[2021-03-04T09:03:29.438536062+02:00][INFO][2192 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Remote)
[2021-03-04T09:03:29.438976253+02:00][INFO][2192 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(623)] model hash updater starting
[2021-03-04T09:03:29.439234853+02:00][INFO][2192 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Local)
[2021-03-04T09:03:29.439388426+02:00][INFO][2192 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Local)
[2021-03-04T09:03:29.494071306+02:00][INFO][2192 main][ui::http_server][Some("tabnine/ui/src/http_server.rs")][Some(591)] Started config server at 127.0.0.1:5555.
[2021-03-04T09:03:29.494418730+02:00][INFO][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/message_shown_timestamp_default.prefs.json")
[2021-03-04T09:03:29.494600835+02:00][WARN][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T09:03:29.494768772+02:00][INFO][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-04T09:03:29.494895264+02:00][WARN][2192 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T09:03:29.510443441+02:00][INFO][2192 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(256)] experiment id cache duration 12.09813ms
[2021-03-04T09:03:29.715249372+02:00][INFO][2192 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(199)] types cache duration 204.664447ms
[2021-03-04T09:03:29.924079686+02:00][INFO][2192 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(89)] result is Ok(())
[2021-03-04T09:03:29.924858618+02:00][INFO][2192 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(256)] experiment id cache duration 609.941µs
[2021-03-04T09:03:29.925258995+02:00][INFO][2192 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(199)] types cache duration 316.399µs
[2021-03-04T09:03:30.099380509+02:00][INFO][2192 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(89)] result is Ok(())
[2021-03-04T09:03:30.231038560+02:00][INFO][2192 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(628)] model hash updater exiting
[2021-03-04T09:03:30.514557170+02:00][INFO][2192 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/login_happened.prefs.json")
[2021-03-04T09:03:30.514823440+02:00][WARN][2192 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:51.304754697+02:00][INFO][2061 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(38)] token path: /home/denta/.config/TabNine/tabnine.token
[2021-03-04T10:11:51.344774709+02:00][INFO][2061 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(21)] Failed to read token file, user cannot be identified: No such file or directory (os error 2)
[2021-03-04T10:11:51.361877506+02:00][WARN][2061 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(166)] No user token available
[2021-03-04T10:11:51.361999185+02:00][INFO][2061 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(434)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log
[2021-03-04T10:11:51.363291022+02:00][INFO][2061 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-04T10:11:51.393217993+02:00][WARN][2061 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:51.393351794+02:00][INFO][2061 main][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(93)] config is BootstrapConfig { bootstrapper_enabled: true, api_endpoint: "https://api.tabnine.com/experiments/register/v2", registration_interval_seconds: 3600, bark_interval_seconds: 2, force_version: None }
[2021-03-04T10:11:51.417574378+02:00][INFO][2061 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:51.672684964+02:00][INFO][2061 main][auth::user::token_watcher][Some("shared/auth/src/user/token_watcher.rs")][Some(65)] Start watching token file
[2021-03-04T10:11:51.958948023+02:00][WARN][2061 main][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(564)] Cannot run local model: FMA instructions not supported

[2021-03-04T10:11:51.959415865+02:00][INFO][2061 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(64)] Initializing watchdog client...
[2021-03-04T10:11:52.146626707+02:00][INFO][2061 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T10:11:52.148967386+02:00][INFO][2061 bs_main_autobarker][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T10:11:52.158119733+02:00][INFO][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T10:11:52.186444583+02:00][INFO][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/tabnine_version_3.3.82.prefs.json")
[2021-03-04T10:11:52.225954020+02:00][INFO][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-04T10:11:52.226177017+02:00][WARN][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:52.226286609+02:00][INFO][2061 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(199)] trying to start tabnine version Some("3.3.82")
[2021-03-04T10:11:52.226448267+02:00][INFO][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T10:11:52.226724502+02:00][INFO][2061 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(289)] spawning tabnine with args ["/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine", "--client", "jupyter", "--log-file-path", "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log", "--no_bootstrap"]
[2021-03-04T10:11:52.227472773+02:00][INFO][2061 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(293)] started tabnine process pid 2089 version Some("3.3.82"), desired version was Some("3.3.82") this version is Some("3.3.82")
[2021-03-04T10:11:52.227947459+02:00][INFO][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T10:11:52.230135246+02:00][INFO][2089 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(38)] token path: /home/denta/.config/TabNine/tabnine.token
[2021-03-04T10:11:52.230422502+02:00][INFO][2089 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(21)] Failed to read token file, user cannot be identified: No such file or directory (os error 2)
[2021-03-04T10:11:52.230595616+02:00][WARN][2089 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(166)] No user token available
[2021-03-04T10:11:52.230693942+02:00][INFO][2089 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(434)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log --no_bootstrap
[2021-03-04T10:11:52.231712252+02:00][INFO][2089 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(64)] Initializing watchdog client...
[2021-03-04T10:11:52.245251469+02:00][INFO][2061 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T10:11:52.367549233+02:00][INFO][2089 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T10:11:52.420129278+02:00][INFO][2061 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-03-04T10:11:52.420378679+02:00][INFO][2089 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-03-04T10:11:52.717290391+02:00][INFO][2061 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:52.808594664+02:00][INFO][2089 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:52.915563797+02:00][INFO][2089 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Local)
[2021-03-04T10:11:52.915627704+02:00][INFO][2089 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Remote)
[2021-03-04T10:11:52.915807013+02:00][INFO][2089 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Local)
[2021-03-04T10:11:52.916115005+02:00][INFO][2089 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Remote)
[2021-03-04T10:11:52.916209415+02:00][INFO][2089 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(623)] model hash updater starting
[2021-03-04T10:11:52.923372347+02:00][INFO][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T10:11:52.923774499+02:00][INFO][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/message_shown_timestamp_default.prefs.json")
[2021-03-04T10:11:52.923882712+02:00][WARN][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:52.923992646+02:00][INFO][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-04T10:11:52.924128652+02:00][WARN][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:53.009874096+02:00][INFO][2061 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:53.050933556+02:00][WARN][2089 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(564)] Cannot run local model: FMA instructions not supported

[2021-03-04T10:11:53.098709028+02:00][INFO][2089 main][frontend::initialize][Some("tabnine/frontend/src/initialize.rs")][Some(5)] Started.
[2021-03-04T10:11:53.121166686+02:00][INFO][2061 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:53.132714994+02:00][INFO][2089 main][auth::user::token_watcher][Some("shared/auth/src/user/token_watcher.rs")][Some(65)] Start watching token file
[2021-03-04T10:11:53.151499536+02:00][INFO][2089 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:53.353685401+02:00][INFO][2061 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:53.353705890+02:00][INFO][2089 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T10:11:53.996291731+02:00][INFO][2061 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(260)] current version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), spawned version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), remote version is Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }
[2021-03-04T10:11:54.046761721+02:00][INFO][2089 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(623)] model hash updater starting
[2021-03-04T10:11:54.047074465+02:00][INFO][2089 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Local)
[2021-03-04T10:11:54.048125719+02:00][INFO][2089 main][ui::http_server][Some("tabnine/ui/src/http_server.rs")][Some(591)] Started config server at 127.0.0.1:5555.
[2021-03-04T10:11:54.048301654+02:00][INFO][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/message_shown_timestamp_default.prefs.json")
[2021-03-04T10:11:54.048408075+02:00][WARN][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:54.048532165+02:00][INFO][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-04T10:11:54.048624489+02:00][WARN][2089 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T10:11:54.049860834+02:00][INFO][2089 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Remote)
[2021-03-04T10:11:54.050600246+02:00][INFO][2089 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Local)
[2021-03-04T10:11:54.051089486+02:00][INFO][2089 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Remote)
[2021-03-04T10:11:54.051175525+02:00][INFO][2089 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(628)] model hash updater exiting
[2021-03-04T10:11:54.053093182+02:00][INFO][2089 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(256)] experiment id cache duration 635.683µs
[2021-03-04T10:11:54.053460908+02:00][INFO][2089 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(199)] types cache duration 237.213µs
[2021-03-04T10:11:54.221948758+02:00][INFO][2061 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(332)] don't need to upgrade
[2021-03-04T10:11:54.257284832+02:00][INFO][2089 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(89)] result is Ok(())
[2021-03-04T10:11:54.258240276+02:00][INFO][2089 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(256)] experiment id cache duration 612.657µs
[2021-03-04T10:11:54.258746895+02:00][INFO][2089 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(199)] types cache duration 422.599µs
[2021-03-04T10:11:54.441185116+02:00][INFO][2089 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(89)] result is Ok(())
[2021-03-04T10:11:55.099101363+02:00][INFO][2089 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/login_happened.prefs.json")
[2021-03-04T10:11:55.099303894+02:00][WARN][2089 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:14.809253236+02:00][INFO][7084 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(38)] token path: /home/denta/.config/TabNine/tabnine.token
[2021-03-04T11:03:14.809533867+02:00][INFO][7084 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(21)] Failed to read token file, user cannot be identified: No such file or directory (os error 2)
[2021-03-04T11:03:14.809616460+02:00][WARN][7084 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(166)] No user token available
[2021-03-04T11:03:14.809686142+02:00][INFO][7084 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(434)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log
[2021-03-04T11:03:14.810122344+02:00][INFO][7084 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-04T11:03:14.810236917+02:00][WARN][7084 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:14.810318430+02:00][INFO][7084 main][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(93)] config is BootstrapConfig { bootstrapper_enabled: true, api_endpoint: "https://api.tabnine.com/experiments/register/v2", registration_interval_seconds: 3600, bark_interval_seconds: 2, force_version: None }
[2021-03-04T11:03:14.810438886+02:00][INFO][7084 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:14.919380557+02:00][INFO][7084 main][auth::user::token_watcher][Some("shared/auth/src/user/token_watcher.rs")][Some(65)] Start watching token file
[2021-03-04T11:03:15.078963856+02:00][WARN][7084 main][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(564)] Cannot run local model: FMA instructions not supported

[2021-03-04T11:03:15.079433062+02:00][INFO][7084 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(64)] Initializing watchdog client...
[2021-03-04T11:03:15.253867216+02:00][INFO][7084 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T11:03:15.255096765+02:00][INFO][7084 bs_main_autobarker][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(191)] Starting watchdog...
[2021-03-04T11:03:15.255130934+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T11:03:15.255728339+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/tabnine_version_3.3.82.prefs.json")
[2021-03-04T11:03:15.256209917+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/bootstrapper_conf.prefs.json")
[2021-03-04T11:03:15.256310840+02:00][WARN][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:15.256372404+02:00][INFO][7084 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(199)] trying to start tabnine version Some("3.3.82")
[2021-03-04T11:03:15.256460445+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T11:03:15.256546180+02:00][INFO][7084 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(289)] spawning tabnine with args ["/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine", "--client", "jupyter", "--log-file-path", "/home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log", "--no_bootstrap"]
[2021-03-04T11:03:15.259274462+02:00][INFO][7084 bs_main_rlp][TabNine::bootstrap_support][Some("tabnine/frontend/src/bootstrap_support.rs")][Some(293)] started tabnine process pid 7113 version Some("3.3.82"), desired version was Some("3.3.82") this version is Some("3.3.82")
[2021-03-04T11:03:15.259611950+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T11:03:15.266653129+02:00][INFO][7113 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(38)] token path: /home/denta/.config/TabNine/tabnine.token
[2021-03-04T11:03:15.266863359+02:00][INFO][7113 main][auth::user::token_holder][Some("shared/auth/src/user/token_holder.rs")][Some(21)] Failed to read token file, user cannot be identified: No such file or directory (os error 2)
[2021-03-04T11:03:15.266930249+02:00][WARN][7113 main][auth::user][Some("shared/auth/src/user/mod.rs")][Some(166)] No user token available
[2021-03-04T11:03:15.266986177+02:00][INFO][7113 main][TabNine][Some("tabnine/frontend/src/main.rs")][Some(434)] /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/binaries/3.3.82/x86_64-unknown-linux-gnu/TabNine --client jupyter --log-file-path /home/denta/Datawiz/3.03.2021/Blog/.venv/lib/python3.8/site-packages/jupyter_tabnine/tabnine.log --no_bootstrap
[2021-03-04T11:03:15.267561993+02:00][INFO][7113 main][watchdog_api::wd_client][Some("shared/wd_api/src/wd_client.rs")][Some(64)] Initializing watchdog client...
[2021-03-04T11:03:15.273758126+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T11:03:15.273967062+02:00][INFO][7084 watchdog spawning][process_setup::command_ext][Some("shared/process_setup/src/command_ext.rs")][Some(24)] child exited with status exit code: 0
[2021-03-04T11:03:15.564865255+02:00][INFO][7084 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:15.804332420+02:00][INFO][7113 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:15.883218591+02:00][INFO][7113 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Remote)
[2021-03-04T11:03:15.884023577+02:00][INFO][7113 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Local)
[2021-03-04T11:03:15.884098086+02:00][INFO][7113 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Remote)
[2021-03-04T11:03:15.884220899+02:00][INFO][7113 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(623)] model hash updater starting
[2021-03-04T11:03:15.884337358+02:00][INFO][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T11:03:15.885302867+02:00][INFO][7113 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Local)
[2021-03-04T11:03:15.886471009+02:00][INFO][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/message_shown_timestamp_default.prefs.json")
[2021-03-04T11:03:15.886735968+02:00][WARN][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:15.886844479+02:00][INFO][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-04T11:03:15.886923868+02:00][WARN][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:15.889692092+02:00][INFO][7113 main][frontend::initialize][Some("tabnine/frontend/src/initialize.rs")][Some(5)] Started.
[2021-03-04T11:03:15.956096322+02:00][WARN][7113 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(564)] Cannot run local model: FMA instructions not supported

[2021-03-04T11:03:15.994994452+02:00][INFO][7113 main][auth::user::token_watcher][Some("shared/auth/src/user/token_watcher.rs")][Some(65)] Start watching token file
[2021-03-04T11:03:15.995996193+02:00][INFO][7113 main][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:16.005757993+02:00][INFO][7084 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:16.202976356+02:00][INFO][7113 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:16.204546186+02:00][INFO][7084 thread][global_config::global_config][Some("shared/global_config/src/global_config.rs")][Some(676)] config path: /home/denta/.config/TabNine/tabnine_config.json
[2021-03-04T11:03:16.881571244+02:00][INFO][7113 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Local)
[2021-03-04T11:03:16.881897965+02:00][INFO][7113 lunar settings update thread (Local)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Local)
[2021-03-04T11:03:16.882487060+02:00][INFO][7113 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(510)] lunar settings update thread started (Remote)
[2021-03-04T11:03:16.882763674+02:00][INFO][7113 lunar settings update thread (Remote)][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(516)] lunar settings update thread exiting (Remote)
[2021-03-04T11:03:16.882819794+02:00][INFO][7113 main][ui::http_server][Some("tabnine/ui/src/http_server.rs")][Some(591)] Started config server at 127.0.0.1:5555.
[2021-03-04T11:03:16.883108749+02:00][INFO][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/message_shown_timestamp_default.prefs.json")
[2021-03-04T11:03:16.883259417+02:00][WARN][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:16.883433525+02:00][INFO][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/experiment_converged_default.prefs.json")
[2021-03-04T11:03:16.883559564+02:00][WARN][7113 main][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T11:03:16.887610213+02:00][INFO][7113 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(623)] model hash updater starting
[2021-03-04T11:03:16.889192543+02:00][INFO][7113 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(256)] experiment id cache duration 427.211µs
[2021-03-04T11:03:16.889473234+02:00][INFO][7113 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(199)] types cache duration 204.347µs
[2021-03-04T11:03:16.956419826+02:00][INFO][7113 model hash update thread][lunar_connection][Some("tabnine/lunar_connection/src/lib.rs")][Some(628)] model hash updater exiting
[2021-03-04T11:03:17.071170507+02:00][INFO][7113 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(89)] result is Ok(())
[2021-03-04T11:03:17.072018613+02:00][INFO][7113 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(256)] experiment id cache duration 666.098µs
[2021-03-04T11:03:17.072446522+02:00][INFO][7113 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(199)] types cache duration 328.191µs
[2021-03-04T11:03:17.102449512+02:00][INFO][7084 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(260)] current version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), spawned version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), remote version is Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }
[2021-03-04T11:03:17.304971658+02:00][INFO][7113 main][analytics::db::real][Some("shared/analytics/src/db.rs")][Some(89)] result is Ok(())
[2021-03-04T11:03:17.445556025+02:00][INFO][7084 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(332)] don't need to upgrade
[2021-03-04T11:03:17.890237076+02:00][INFO][7113 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/login_happened.prefs.json")
[2021-03-04T11:03:17.890428290+02:00][WARN][7113 thread][defaults][Some("shared/defaults/src/lib.rs")][Some(115)] had an error deserializing object Os { code: 2, kind: NotFound, message: "No such file or directory" }
[2021-03-04T12:03:17.445966512+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T12:03:17.446377445+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/tabnine_version_3.3.82.prefs.json")
[2021-03-04T12:03:17.446902077+02:00][INFO][7084 bs_main_rlp][defaults][Some("shared/defaults/src/lib.rs")][Some(99)] getting value from path Ok("/home/denta/.config/TabNine/defaults/active_experiment.prefs.json")
[2021-03-04T12:03:19.289423197+02:00][INFO][7084 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(260)] current version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), spawned version is Some(Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }), remote version is Version { major: 3, minor: 3, patch: 82, pre: [], build: [] }
[2021-03-04T12:03:19.485450291+02:00][INFO][7084 bs_main_rlp][bootstrapper::bootstrapper_lib][Some("tabnine/bootstrapper/src/bootstrapper_lib.rs")][Some(332)] don't need to upgrade
